pkg/netutil/net-systemd.go

```diff
@@ -0,0 +1,50 @@
+package netutil
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	systemdDbus "github.com/coreos/go-systemd/dbus"
+)
+
+// CheckForbiddenService assess that the systemd services which break the network are not enabled
+func CheckForbiddenService() error {
+	forbidden_services := []string{"nm-cloud-setup.service", "nm-cloud-setup.timer"}
+
+	// Verify if we are running systemd. If not, we assume the forbidden services are not present
+	_, err := os.Stat("/usr/lib/systemd")
+	if err != nil {
+		return nil
+	}
+
+	connection, err := systemdDbus.NewSystemdConnection()
+	defer connection.Close()
+	if err != nil {
+		return err
+	}
+
+	// Check if forbidden services are enabled
+	unitsStatus, err := connection.ListUnitsByNames(forbidden_services)
+	if err != nil {
+		// ListUnitsByNames method requires systemd version > 230. If that's the case, we use a slower method
+		if strings.Contains(fmt.Sprint(err), "Unknown method 'ListUnitsByNames' or interface 'org.freedesktop.systemd1.Manager'.") {
```

Comment:
It might be a little cleaner to call `err.Error()` here rather than passing the interface down.